src/org/pahappa/systems/registrationapp/services/UserService.java

```diff
@@ -1,4 +1,41 @@
 package org.pahappa.systems.registrationapp.services;
 
+import org.pahappa.systems.registrationapp.models.User;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+
 public class UserService {
+    public static List<User> users = new ArrayList<>();
+
```

Comment:
The users list is defined as a static variable but is re-initialized in the constructor. This approach can lead to confusion. If users is meant to be a shared list across all instances of UserService **which is not right**, it should remain static and not be re-initialized in the constructor.

So change access modifiers of this users field to private and use it as an instance field